### DISCUSSION TITLE
请教一下实现 macOS 自带输入法的英语单词功能，最简单的办法是?

### DISCUSSION THREAD

**Discussion by pdog18** [rime/home#1284] [Q&A]
我只需要 Rime 可以像 macOS 一样，输入
1.  `[amazi]` 的时候提示`「amazing」`
2. `[choos]` 的时候提示`「choose」`
3. `[thi]` 的时候提示 「体会」但是 `[this]`的时候，selected 候选词是 `「this」`

就是「差不多在输入码达到`一定程度`的时候才会提示单词，而不是输出一个 `[ca]` 就会提示`「cat」`」
最简单的实现方案是怎样的？我查看了 `easy-en` 与 `rime-melt` 似乎无法实现，想来请教下，如果只是为了实现和原生输入法类似的效果，有没有更简单的方法呢？（主要是小白可以轻松启用）

**Comment by pdog18** [comment]
我尝试直接用外挂词库的方式，发现自己想得过于简单了，完全不可行。
如果外挂一个简单的，内部是一组「单词  单词输入码  字频」的 `en.dict.yaml` ，会导致这种结果发生:
输入 `[easy]`候选词会只有一个`「easy」`，而不是期望的`「easy、恩爱使用、恩爱、而、恩」`，特别特别奇怪。

**Comment by pdog18** [comment]
简单看了 [RimeWithSchemata](https://github.com/rime/home/wiki/RimeWithSchemata#rime-with-schemata)，感觉要完美支持英文单词似乎要 `Segmentors` 的支持？

**Reply by pdog18** [reply]
> 简单看了 [RimeWithSchemata](https://github.com/rime/home/wiki/RimeWithSchemata#rime-with-schemata)，感觉要完美支持英文单词似乎要 `Segmentors` 的支持？

感觉这想法不靠谱，总不能 rime 内核内置一套英文单词的码表吧，不然怎么分段啊

**Comment by pdog18** [comment]
我对现在的 `easy-en` 、`rime-melt` 使用英文单词的方案感到非常困惑。

其他方案（如 `pinyin_simp`）在依赖这两个方案时，在注明了依赖之后 `schema/dependencies:` ，仍然需要在主方案(如`pinyin_simp`)中继续声明 `table_translator` ,例如:
```
  engine/translators/+:
   - table_translator@melt_eng          
 


 melt_eng:
   dictionary: melt_eng     
   enable_sentence: false   
   enable_user_dict: false  
   initial_quality: 1.1     
   comment_format:          
     - xform/.*//           
  ```
  
  这样的代码看起来很无厘头, 既然你依赖了目标方案, 还需要写一个 `table_translator` 干嘛?既然一定要写一个 `table_translator` 那为什么不直接用 `table_translator` 还要依赖别的方案是干嘛?
  
可是事实两者必须都需要, 否则英文单词则无效. 太反直觉了. 也搞不清楚为什么会这样.

**Reply by HF706** [reply]
我也非常需要这个功能，但不知道怎么做。我找了哈利路亚输入法代替了。[哈利路亚输入法](https://github.com/dongyuwei/hallelujahIM)

**Reply by pdog18** [reply]
你使用 markdown 的时候应该是 
`[显示的名字](url链接)` 
不然别人点击后跳转到一个奇怪的 url

https://github.com/dongyuwei/hallelujahIM

**Reply by HF706** [reply]
感谢指正
